### DISCUSSION TITLE
Using M3/M4 to decrease/increase screen brightness code changes

### DISCUSSION THREAD

**Discussion by dparson55** [seerge/g-helper#322] [Q&A] [answered]
Hi all, I'm another person wanting to remap M3 and M4 to something new and wondered if someone could help me create actions for decreasing or increasing screen brightness.

I've been studying the code and used to do some light windows programming 20 years ago but never ventured into the world of integrating with other apps via acpi and wmi and using hid devices.  But I am remembering a lot of the basics and can build ghelper locally and have been stepping through it to watch it work, but I'm scratching the surface of what is going on still.

I see a delegate in Extra.SetKeyCombo (https://github.com/seerge/g-helper/blob/main/app/Extra.cs#L52) saves whatever a user enters as a custom action in to the app.config but haven't been able to find a keyboard handler that reads the app.config to perform the action that was saved.  Does anyone know where this is being done?  I'm wondering too if I'll need to create some constants in https://github.com/seerge/g-helper/blob/main/app/ASUSWmi.cs for increasing and decreasing the brightness and am at a loss for how they would be determined.  I enabled wmi tracing in the event log and hit the brighness increase and decrease keys on my keyboard and did see wmi calls being made, but it's so cryptic that I don't see what wmi values are being set.  Is there a method to determine these?

**Comment by vocaloidict** [comment]
I'd also like this function - M3 to brightness down, M4 to brightness up.

Alternatively if there's a way to bind key combinations that would work too - e.g. Fn+F7 is brightness down

**Comment by seerge** [comment, marked as answer]
@dparson55 @vocaloidict added

[GHelper.zip](https://github.com/seerge/g-helper/files/11677894/GHelper.zip)

**Reply by vocaloidict** [reply]
Wow that was fast. Works perfectly, thank you!

**Reply by dparson55** [reply]
Thank you!
